keel-sql/src/main/resources/db/changelog/20201016-create-lifecycle-events.yml

```diff
@@ -0,0 +1,151 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-lifecycle-events
+      author: emjburns
+      changes:
+        - createTable:
+            tableName: lifecycle
+            columns:
+              - column:
+                  name: uid
+                  type: char(26)
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: scope
+                  type: varchar(50)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: ref
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: artifact_version
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: type
+                  type: varchar(50)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: id
+                  type: varchar(100)
+                  constraints:
+                    nullable: false
```

Comment:
I'm getting a little confused what these fields refer to. Is `type` the type of event? What is `id`? Is that the bakery id? Is the table structure too fine grained in that case?

Reply:
type is the type of event (`BAKE` for now). Id is a caller-generated string that ties multiple events together. It is what we group by to summarize. So for example, the bake "id" right now is `bake-$version`. That allows us to group all the events for that bake into a summary and allows you to have multiple of the same type of event in different summaries (like, for example, a rebake because the first bake failed).

I'm in general a bit confused about what to put in table columns vs what to shove in a json blob. I've been thinking about it like: everything I might want to query or filter on I should put in a column. Does more columns = slower/more overhead? Is there a downside to more columns?